pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -144,9 +150,10 @@ func ResolvePipelineRun(getTask resources.GetTask, getResource resources.GetReso
 
 		// Get all the resources that this task will be using, if any
 		inputs, outputs := getPipelineRunTaskResources(pt.Name, pr)
-		rtr, err := resources.ResolveTaskResources(&t.Spec, t.Name, inputs, outputs, getResource)
+		spec := t.TaskSpec()
+		rtr, err := resources.ResolveTaskResources(&spec, t.TaskMetadata().Name, inputs, outputs, getResource)
```

Comment:
nit: Could we have a `t.Name()` function here ? (kind-of a shortcut so, not that important)